code/modules/projectiles/guns/projectile.dm

```diff
@@ -149,7 +149,8 @@
 		if(gun_flags &CHAMBERSPENT)
 			refuse += AC
 		else
-			AC.forceMove(usr.loc) //Eject casing onto ground or closet you're inside.
+			var/mob/M = get_holder_of_type(src, /mob/)
+			AC.forceMove(ismob(M) ? M.loc : get_turf(src.loc)) //special forceMove because this proc hate user so much it breaks if you try to use mob/user in arg
```

Comment:
Was `user` passed to this proc before? That might be why it wasn't working.

Reply:
no, it isn't passed, it isn't defined with one too, but this proc is often used in gun code(and subsequently gun procs) as a check
writing this would avoid accidently breaking innumerous things